plugins/defaultplugins/srplugin/vppcalls/srv6.go

```diff
@@ -0,0 +1,510 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vppcalls
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/measure"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/sr"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/srv6"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+)
+
+// Constants for behavior function hardcoded into VPP (there can be also custom behavior functions implemented as VPP plugins)
+// Constants are taken from VPP's vnet/srv6/sr.h (names are modified to Golang from original C form in VPP code)
+const (
+	BehaviorEnd    uint8 = iota + 1 //Behavior of simple endpoint
+	BehaviorX                       //Behavior of endpoint with Layer-3 cross-connect
+	BehaviorT                       //Behavior of endpoint with specific IPv6 table lookup
+	BehaviorDfirst                  //Unused. Separator in between regular and D
+	BehaviorDX2                     //Behavior of endpoint with decapulation and Layer-2 cross-connect (or DX2 with egress VLAN rewrite when VLAN notzero - not supported this variant yet)
+	BehaviorDX6                     //Behavior of endpoint with decapsulation and IPv6 cross-connect
+	BehaviorDX4                     //Behavior of endpoint with decapsulation and IPv4 cross-connect
+	BehaviorDT6                     //Behavior of endpoint with decapsulation and specific IPv6 table lookup
+	BehaviorDT4                     //Behavior of endpoint with decapsulation and specific IPv4 table lookup
+	BehaviorLast                    //seems unused, note in VPP: "Must always be the last one"
+)
+
+// Constants for steering type
+// Constants are taken from VPP's vnet/srv6/sr.h (names are modified to Golang from original C form in VPP code)
+const (
+	SteerTypeL2 uint8 = iota*2 + 2
+	SteerTypeIPv4
+	SteerTypeIPv6
+)
+
+// Constants for operation of SR policy modify binary API method
+const (
+	AddSRList            uint8 = iota + 1 //Add SR List to an existing SR policy
+	DeleteSRList                          //Delete SR List from an existing SR policy
+	ModifyWeightOfSRList                  //Modify the weight of an existing SR List
+)
+
+// SRv6Calls is API boundary for vppcall package access, introduced to properly test code dependent on vppcalls package
+type SRv6Calls interface {
+	// AddLocalSid adds local sid given by <sidAddr> and <localSID> into VPP
+	AddLocalSid(sidAddr net.IP, localSID *srv6.LocalSID, swIfIndex ifaceidx.SwIfIndex, log logging.Logger, vppChan VPPChannel, stopwatch *measure.Stopwatch) error
```

Comment:
Some generic parameters like logger, vpp channel and stopwatch can be removed from this interface definition and be injected into srv6calls through its constructor. This will make the entire interface smaller and eliminate the requirements for these parameters in test fakes.

Reply:
Fixed in 0b3711cc2e817413fdf7bcd64b9cbf1cfe61f18d .

I also tried to move out vppChan, but i either ended up to with creation and closing of channel in different levels of component hierarchy or i couldn't create vppCalls implementation in one component hierarchy level or i hit the wall by unable to create mock of struct (GoVppmux interface methods return struct instead of interfaces). In the end i decided to not to move it out of API for now.